test/Dialect/FIRRTL/SFCTests/data-taps-xmr.fir

```diff
@@ -117,3 +117,60 @@ circuit Top : %[[
 ; CHECK:         ChildWrapper unsigned_0 (
 ; CHECK-NEXT:      .[[ChildWrapper_boredPort]] (Top.int_0.io_out)
 ; CHECK:       endmodule
+
+; // -----
+
+; XMR references should work through 'when's.
+; https://github.com/llvm/circt/issues/4334
+circuit Top : %[[
+  {
+    "class":"sifive.enterprise.grandcentral.DataTapsAnnotation",
+    "keys":[
+      {
+        "class":"sifive.enterprise.grandcentral.ReferenceDataTapKey",
+        "source":"~Top|Top>val",
+        "sink":"~Top|Top/mid:Middle/leaf:Leaf>tap"
+      },
+      {
+        "class":"sifive.enterprise.grandcentral.ReferenceDataTapKey",
+        "source":"~Top|Top/mid:Middle/leaf:Leaf>val",
+        "sink":"~Top|Top>tap"
+      }
+    ]
+  }
+]]
+  module Leaf :
+    input c: UInt<1>
+
+    when c:
+      wire val : UInt<3>
+      val <= UInt(2)
+
+      ; Tap Top's val
+      wire tap : UInt<4>
+      tap is invalid
+
+  module Middle :
+    input c: UInt<1>
+    
+    when c:
+      inst leaf of Leaf
+      leaf.c <= c
+
+  module Top :
+    input c: UInt<1>
+    output o: UInt
+
+    wire val : UInt<3>
+    val <= UInt(1)
+
+    inst mid of Middle
+    mid.c <= c
+
+    o is invalid
+
+    ; Tap Leaf's val
+    when c:
+      wire tap : UInt
+      tap is invalid
+      o <= tap
```

Comment:
test Check missing ?

Reply:
Good call, thanks! It working at all is success but should check the expected output too, agreed.